### DISCUSSION TITLE
Does the Python-chess library work in a 32-bit installation of Python 3.12?

### DISCUSSION THREAD

**Discussion by VladimirMishka** [niklasf/python-chess#1059] [Q&A] [answered]
Would someone please be so kind as to tell me if the python-chess library will work under a 32-bit installation of Python 3.12 in Windows 11? If so, are there any limitations?

I am working on a teaching app for 3-check chess and one possibility is to use Delphi and the Python4Delphi package (I am very familiar with Delphi, but new to other possible solutions such as a Flask app). I have the python-chess library working with a 64-bit installation of Python in Python4Delphi. However, certain other third-party Delphi components that I would like to use in the project (e.g., TSVGIconImage) only seem to be working when compiled into a 32-bit application (I consistently get run-time errors when I compile under 64-bit).

Thank you very much for your help.

**Comment by niklasf** [comment, marked as answer]
The core module (`import chess`) of this library should be completely platform and architecture independent.

There are some caveats for other modules:

* `chess.gaviota`, `chess.syzygy` and `chess.polyglot` map tablebase and opening book files into memory. The maximum size may be limited on 32 bit platforms.
* `chess.engine` spawns processes. So the operating system needs to support that, and the engine executables need to be compatible.
